crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -536,9 +544,26 @@ impl ScheduleGraph {
                             densely_chained: true
                         }
                     };
+                    let more_than_one_entry = config_iter.len() > 0;
+                    let anonymous_set = if more_than_one_entry && !collective_conditions.is_empty()
```

Comment:
We never actually add the collective conditions anywhere so run_if doesn't work at all here.

Reply:
Fixed in my commit below